AI/Nullkiller/Pathfinding/AINodeStorage.cpp

```diff
@@ -1536,7 +1536,7 @@ bool AIPath::containsHero(const CGHeroInstance * hero) const
 
 uint64_t AIPath::getTotalArmyLoss() const
 {
-	return armyLoss + targetObjectArmyLoss;
+	return armyLoss > targetObjectArmyLoss ? armyLoss : targetObjectArmyLoss;
```

Comment:
Army loss should be cumulative. The issue of double loss counting may be because we simulate battle with same monster twice. You can check how it will work with guarded dragon utopia for instance when guard strength is close to utopia strength.

Reply:
How do I avoid checking the battle with the same enemy twice? I've looked at logs and usually it was either armyLoss or targetObjectArmyLoss that contained a value. In rare occassions it was both and in some of those both had the exact same value, which indicated the double-checking of the same fight.

Reply:
Maybe I can take a look onto it if you provide me a small simple map where it is visible so I can debug.

Reply:
Or I can take a look myself. Most likely it is fighting monster by cluster unlock